tests/pass/intrinsics-x86-sse2.rs

```diff
@@ -70,6 +70,16 @@ mod tests {
         }
         test_mm_avg_epu16();
 
+        #[target_feature(enable = "sse2")]
+        unsafe fn test_mm_madd_epi16() {
+            let a = _mm_setr_epi16(1, 2, 3, 4, 5, 6, 7, 8);
+            let b = _mm_setr_epi16(9, 10, 11, 12, 13, 14, 15, 16);
+            let r = _mm_madd_epi16(a, b);
+            let e = _mm_setr_epi32(29, 81, 149, 233);
+            assert_eq_m128i(r, e);
```

Comment:
Can you add a test that exploits the fact that the i16 become i32 before being multiplied?